web3.go

```diff
@@ -109,7 +109,7 @@ func CallConstantFunction(ctx context.Context, client Client, myabi abi.ABI, add
 			return nil, err
 		}
 		for i, o := range out {
-			out[i] = reflect.ValueOf(o).Elem()
+			out[i] = fmt.Sprintf("%v", reflect.ValueOf(o).Elem())
```

Comment:
Don't we need to preserve the type?

Reply:
I think by default it's pointer

Reply:
`Elem()` returns a `reflect.Value`. `Interface()` should work:
```suggestion
			out[i] = fmt.Sprintf("%v", reflect.ValueOf(o).Elem())
			out[i] = reflect.ValueOf(o).Elem().Interface()
```